File: ATWeb/models/user-work-history.js
'use strict';

module.exports = (sequelize, DataTypes) => {
    const Op = require('sequelize').Op;
    const UserWorkHistory = sequelize.define('UserWorkHistory',
        {
            periodStart: { type: DataTypes.DATEONLY, allowNull: false },
            periodEnd: { type: DataTypes.DATEONLY },
            companyName: { type: DataTypes.STRING(128), allowNull: false },
            companyAddress: { type: DataTypes.STRING(256), allowNull: false },
            position: { type: DataTypes.STRING(50), allowNull: false },
            details: { type: DataTypes.STRING(512) },
        }, {
            timestamps: true,
            scopes: {
                byUserId: function (userId) {
                    return {
                        where: { userId: { [Op.like]: userId } },
                        order: [['periodStart', 'ASC']]
                    };
                }
            }
        });

    return UserWorkHistory;
};

File: ATCommon/Models/UserWorkHistory.cs
using System;

namespace ATCommon.Models
{
    public class UserWorkHistory
    {
        public string userId { get; set; }
        public DateTime periodStart { get; set; }
        public DateTime periodEnd { get; set; }
        public string companyName { get; set; }
        public string companyAddress { get; set; }
        public string position { get; set; }
        public string details { get; set; }
    }
}
